src/python.rs

```diff
@@ -302,6 +386,49 @@ impl<'p> Python<'p> {
         unsafe { PyObject::from_borrowed_ptr(self, ffi::Py_NotImplemented()) }
     }
 
+    /// Gets the running Python interpreter version as a string.
+    ///
+    /// This is a wrapper around the ffi call Py_GetVersion.
+    ///
+    /// # Example
+    /// ```rust
+    /// # use pyo3::Python;
+    /// Python::with_gil(|py| {
+    ///     // The full string could be, for example:
+    ///     // "3.0a5+ (py3k:63103M, May 12 2008, 00:53:55) \n[GCC 4.2.3]"
+    ///     assert!(py.version().starts_with("3."));
+    /// });
+    /// ```
+    pub fn version(self) -> &'static str {
+        unsafe {
+            CStr::from_ptr(ffi::Py_GetVersion() as *const c_char)
```

Comment:
I'm not sure treating the result of `Py_GetVersion` as ` `&'static str` is safe. Because of the way it's implemented, if you have a thread calling `version()` at the same time as another thread is attempting to read the contents of the str, you have a data race: https://github.com/python/cpython/blob/master/Python/getversion.c

Is there a reason it wouldn't work to give it the `'py` lifetime?

Reply:
Ah, yes the claim in the docs it was "static storage" misled me. Nice catch - I'll change the lifetime 👍